app/controllers/schools/change_schools_controller.rb

```diff
@@ -0,0 +1,63 @@
+module Schools
+  class InaccessibleSchoolError < StandardError; end
+
+  class ChangeSchoolsController < Schools::BaseController
+    before_action :ensure_in_app_school_changing_enabled?, only: :create
+
+    rescue_from Schools::InaccessibleSchoolError, with: :access_denied
+
+    def show
+      @schools = Bookings::School.ordered_by_name.where(urn: urns)
+      @change_school = Schools::ChangeSchool.new(id: @current_school.id)
+    end
+
+    def create
+      new_school = Bookings::School.find(change_school_params[:id])
+
+      raise Schools::InaccessibleSchoolError unless user_has_role_at_school?(current_user.sub, new_school.urn)
+
+      session[:urn]         = new_school.urn
+      session[:school_name] = new_school.name
+
+      redirect_to schools_dashboard_path
+    end
+
+  private
+
+    def ensure_in_app_school_changing_enabled?
+      enabled = [
+        Rails.configuration.x.dfe_sign_in_api_enabled,
+        Rails.configuration.x.dfe_sign_in_api_school_change_enabled
+      ].all?
+
+      head(:forbidden) unless enabled
+    end
+
+    def change_school_params
+      params.require(:schools_change_school).permit(:id)
+    end
+
+    def access_denied
+      redirect_to schools_errors_inaccessible_school_path
+    end
+
+    def urns
+      organisations.urns
+    end
+
+    def organisations
+      @organisations ||= Schools::DFESignInAPI::Organisations.new(current_user.sub)
+    end
+
+    def user_has_role_at_school?(user_uuid, urn)
+      Schools::DFESignInAPI::Roles.new(user_uuid, urn).has_school_experience_role?
```

Comment:
This is currently ignoring the feature flag for whether the role check is enabled?

Reply:
There's a `before_action` in the controller that calls `ensure_in_app_school_changing_enabled?`, I think that should do it?

Reply:
Sorry, I'm referring to the role check feature, not the change school feature

If change school is enabled this code will perform role checks - even if the role check feature is disabled

Reply:
Ah, misunderstood the first comment